Modules/Applications/AppDomainTransform/app/otbDomainTransform.cxx

```diff
@@ -0,0 +1,453 @@
+/*=========================================================================
+
+  Program:   ORFEO Toolbox
+  Language:  C++
+  Date:      $Date$
+  Version:   $Revision$
+
+
+  Copyright (c) Centre National d'Etudes Spatiales. All rights reserved.
+  See OTBCopyright.txt for details.
+
+
+  This software is distributed WITHOUT ANY WARRANTY; without even
+  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+  PURPOSE.  See the above copyright notices for more information.
+
+  =========================================================================*/
+
+#include "otbWrapperApplication.h"
+#include "otbWrapperApplicationFactory.h"
+#include "otbWaveletImageFilter.h"
+#include "otbWaveletInverseImageFilter.h"
+#include "otbWaveletGenerator.h"
+
+#include <itkConfigure.h>
+#include <itkForwardFFTImageFilter.h>
+#include <itkInverseFFTImageFilter.h>
+#include <itkUnaryFunctorImageFilter.h>
+#include <itkFFTShiftImageFilter.h>
+
+namespace otb
+{
+namespace Wrapper
+{
+
+template< class TInput, class TOutput>
+class FromComplexPixel
+{
+public:
+  FromComplexPixel ( ) {  };
+
+  ~FromComplexPixel( ) {  };
+
+  bool operator!=( const FromComplexPixel & ) const
+    {
+    return false;
+    }
+  bool operator==( const FromComplexPixel & other ) const
+    {
+    return !(*this != other);
+    }
+  inline TOutput operator( )( const TInput & A ) const
+    {
+    TOutput out;
+    out.SetSize(2);
+    out[0] = A.real();
+    out[1] = A.imag();
+    return out;
+    }
+};
+
+template< class TInput, class TOutput>
+class ToComplexPixel
+{
+public:
+  ToComplexPixel ( ) { };
+
+  ~ToComplexPixel( ) { };
+
+  bool operator!=( const ToComplexPixel & ) const
+    {
+    return false;
+    }
+  bool operator==( const ToComplexPixel & other ) const
+    {
+    return !(*this != other);
+    }
+  inline TOutput operator( )( const TInput & A ) const
+    {
+    TOutput out(A[0], A[1]);
+    return out;
+    }
+};
+
+class DomainTransform : public Application
+{
+public:
+  /** Standard class typedefs. */
+  typedef DomainTransform     Self;
+  typedef Application                   Superclass;
+  typedef itk::SmartPointer<Self>       Pointer;
+
+  typedef itk::SmartPointer<const Self> ConstPointer;
+  typedef float TInputPixel;
+  typedef float TOutputPixel;
+
+  /** Standard macro */
+  itkNewMacro(Self);
+
+  itkTypeMacro(Self, otb::Application);
+
+private:
+  void DoInit() ITK_OVERRIDE
+    {
+    SetName("DomainTransform");
+    SetDescription("Domain Transform application for wavelet and fourier");
+
+    // Documentation
+    SetDocName("DomainTransform");
+    SetDocLongDescription("Domain Transform application for wavelet and fourier");
+    SetDocLimitations("None");
```

Comment:
Important limitation to document : this application is not streamed, BEWARE of large images